friendly/src/main/java/com/google/sps/Email.java

```diff
@@ -0,0 +1,42 @@
+package com.google.sps;
+
+/**
+ * Represents a User's email
+ */
+public class Email {
+    private static final String MESSAGE_CONSTRAINTS = "Emails must be a valid gmail address";
+    private static final String VALIDATION_REGEX = "[A-Za-z0-9.]+@gmail.com";
+    private final String email;
+
+    public Email(String email) throws InvalidNameException{
+        if (!isValidEmail(email)) {
+            throw new InvalidNameException(MESSAGE_CONSTRAINTS);
+        }
+        this.email = email;
+    }
+
+    public static boolean isValidEmail(String email) {
+        return email.matches(VALIDATION_REGEX);
+    }
+
+    public String getEmail() {
+        return this.email;
+    }
+
+    @Override
+    public String toString() {
```

Comment:
Why do you need both getEmail and toString function?

Reply:
getEmail() is the standard getter method for a private variable
toString() is overriding the toString() method in the parent class

Though they both return the same thing, it is more intuitive to call getEmail() when we want to get a private variable than toString(), so we need to getEmail() method; we need the toString() method also as it is implicitly called by some functions in the Java infrastructure

Reply:
Noted, thanks for explaining!